File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/objects/SLPrimObjectDisplayInfo.java
package com.lumiyaviewer.lumiya.slproto.objects;

import com.google.common.base.Strings;

public class SLPrimObjectDisplayInfo extends SLObjectDisplayInfo {
    public final int localID;
    public final boolean payable;
    public final boolean touchable;

    public SLPrimObjectDisplayInfo(SLObjectInfo objectInfo, float f) {
        super(objectInfo.localID, objectInfo.nameKnown ? Strings.nullToEmpty(objectInfo.name) : null, f, objectInfo.hierLevel);
        this.localID = objectInfo.localID;
        this.touchable = objectInfo.isTouchable();
        this.payable = objectInfo.isPayable() || objectInfo.saleType != 0;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/objects/SLPrimObjectDisplayInfo.kt
package com.lumiyaviewer.lumiya.slproto.objects

import com.google.common.base.Strings

open class SLPrimObjectDisplayInfo(objectInfo: SLObjectInfo, distance: Float) :
    SLObjectDisplayInfo(objectInfo.localID, if (objectInfo.nameKnown) Strings.nullToEmpty(objectInfo.name) else null, distance, objectInfo.hierLevel) {
    @JvmField val localID: Int = objectInfo.localID
    @JvmField val touchable: Boolean = objectInfo.isTouchable
    @JvmField val payable: Boolean = objectInfo.isPayable || objectInfo.saleType.toInt() != 0
}
